File: 04.Simulation模拟/leetcode_2840_checkStrings.py
from collections import Counter

class Solution:
    def checkStrings(self, s1: str, s2: str) -> bool:
        return Counter(s1[::2])==Counter(s2[::2]) and Counter(s1[1::2])==Counter(s2[1::2])

if __name__ == "__main__":
    s = Solution()
    s1, s2 = "abcdba", "cabdab"
    print(s.checkStrings(s1, s2))

File: 04.Simulation模拟/leetcode_2840_checkStrings.cpp
#include <iostream>
#include <string>
#include <unordered_map>
using namespace std;

class Solution {
public:
    bool checkStrings(string s1, string s2) {
        unordered_map<char, int> mp0, mp1;
        for (int i=0; i<s1.size(); i++){
            if (i&1){
                mp1[s1[i]]++;
                mp1[s2[i]]--;
            }
            else{
                mp0[s1[i]]++;
                mp0[s2[i]]--;
            }
        }
        for (auto& kv : mp1){
            if (kv.second != 0){
                return false;
            }
        }
        for (auto& kv : mp0){
            if (kv.second != 0){
                return false;
            }
        }
        return true;
    }
};

int main(){
    Solution s;
    string s1 = "abcdba", s2 = "cabdab";
    cout << s.checkStrings(s1, s2) << endl;
}
